### ISSUE TITLE
is there documentation for decorator support

### ISSUE THREAD

**Issue by sacrosanctic** [storybookjs/addon-svelte-csf#70]
it says decorators are supported but i cant find any example code, either for individual stories or globally

https://storybook.js.org/docs/svelte/get-started/whats-a-story
https://github.com/storybookjs/addon-svelte-csf#readme

**Comment by sacrosanctic** [comment]
i found [this](https://storybook.js.org/docs/svelte/writing-stories/decorators) which does mention it but the [story decorator](https://storybook.js.org/docs/svelte/writing-stories/decorators#story-decorators) and [global decorator does](https://storybook.js.org/docs/svelte/writing-stories/decorators#global-decorators) not work

**Comment by j3rem1e** [comment]
Decorator are handled by Storybook, not this addon. "It shoud work" because Storybook is rendering our stories as child of the global/story decorator

**Comment by j3rem1e** [comment]
I can't reproduce any issue with decorators. The official doc seem to be working fine.

The syntax here is not special:

```svelte
<Meta component={Button} decorators={[() => MarginDecorator]}/>
```

and with the new meta export, it's exactly like in the official storybook documentation: 

```svelte
<script context="module">
  import Button from './Button.svelte';

  export const meta = {
    component: Button,
    decorators: [() => MarginDecorator]
  }
</script>
```

Feel free to reopen this issue if you need more details.

**Comment by svallory** [comment]
Documentation regarding how to properly implement decorators would be really appreciated. My decorator works, but I get a typescript type error.

```
Value of type 'typeof DualTheme__SvelteComponent_' has no properties in common with type 'SvelteStoryResult<any, any>'. Did you mean to call it?
```

Also, how would I pass parameters/props to it from the `.storybook/preview.ts` file? Right now I'm handling `parameters` outside of my decorator:

`preview.ts`
```ts
decorators: [
  (story, { parameters }) =>
    parameters.dualTheme !== false ? (DualTheme as any) : story(),
],
```

**Comment by xeho91** [comment]
Aight, I'll reopen for investigation.

**Comment by JReinhold** [comment]
> Also, how would I pass parameters/props to it from the `.storybook/preview.ts` file? Right now I'm handling `parameters` outside of my decorator:
> 
> `preview.ts`
> ```ts
> decorators: [
>   (story, { parameters }) =>
>     parameters.dualTheme !== false ? (DualTheme as any) : story(),
> ],
> ```

I don't fully understand this question, can you elaborate with a use case? @svallory

**Comment by svallory** [comment]
@JReinhold I could have explained it a little better 😅

In the example below, I'm checking for the boolean parameter `dualTheme` in the `preview.ts` configuration file, to decide if I should use the decorator or not.

```
decorators: [
  (story, { parameters }) =>
    parameters.dualTheme !== false ? (DualTheme as any) : story(),
],
```

That's fine for this scenario, but what if I wanted something more complex?

Since I'm returning the default export of the Svelte component as is, I have no idea how I can pass the entire `parameters` object as a prop to my component.

**Comment by rChaoz** [comment]
I totally agree with this. I've had some use cases for decorators as well, and it's very difficult to make it work, when you can only return a Svelte component, without any props. It would be awesome if the Svelte CSF plugin somehow integrated with decorators and had some sort of API to make using them easier.

**Comment by JReinhold** [comment]
@svallory @rChaoz 

Decorators (and the `render` function in regular CSF) supports different return types:

1. component and props:
```js
decorators: [
  (story, { parameters }) =>
    {
      Component: MyComponent,
      props: SomeProps
    }
],
```

2. component only:
```js
decorators: [
  (story, { parameters }) =>
    {
      Component: MyComponent,
    }
],
```
 
3. component directly, which is a shorthand for 2:
 
```js
decorators: [
  (story, { parameters }) => MyComponent
],
```


I realize this might not be fully documented, and we might actually have unintentially removed this detail when we redid the Svelte docs recently, cc @jonniebigodes . I agree it should be documented.



> It would be awesome if the Svelte CSF plugin somehow integrated with decorators and had some sort of API to make using them easier.

@rChaoz any ideas on how such an API could look like would be welcomed. 🙏

**Comment by drewswaycool** [comment]
Do I understand that it is supported and possible and that it is just not documented?

**Comment by fubits1** [comment]
@drewswaycool I can confirm that decorators work. I’m on my phone right now, but here‘s my decorators config (from .storybook/preview.js):

I using a wrapper to set a fallback height for charts, to provide a mock context and to toggle a plugin for debugging.

The only thing I can’t make sense of yet, is why decorators expects an array. 

```{js}
decorators: [
		(Story, { parameters }) => {
			return {
				Component: StorybookPreviewWrapper,
				props: {
					is_render_scan_enabled: false,
					offsetLeft: -20,
					children: Story
				}
			}
		}
	]
```

**Comment by xeho91** [comment]
> The only thing I can’t make sense of yet, is why decorators expects an array. 

It allows you to wrap stories with more than one decorators. Mostly for separation of concern. For example one for auth, another one for i18n provider, etc.
